VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -148,4 +122,110 @@ public void onClickAddComment(@NonNull DiscussionComment comment) {
     public void onClickViewComments(@NonNull DiscussionComment comment) {
         router.showCourseDiscussionComments(getActivity(), comment, discussionThread.isClosed());
     }
+
+    public static class DiscussionResponsesLoader implements
+            InfiniteScrollUtils.PageLoader<DiscussionComment> {
+        private InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback;
+
+        private Context context;
+        private GetCommentListTask getCommentListTask;
+        private GetCommentListTask getEndorsedListTask;
+
+        private int nextPage = 1;
+        private String threadId;
+        private boolean isQuestionTypeThread;
+
+        private ThreadComments unendorsedComments;
+        private boolean isEndorsedFetched = false;
+
+        public DiscussionResponsesLoader(@NonNull Context context,@NonNull String threadId,
+                                         boolean isQuestionTypeThread) {
+            this.context = context;
+            this.threadId = threadId;
+            this.isQuestionTypeThread = isQuestionTypeThread;
+        }
+
+        @Override
+        public void loadNextPage(@NonNull InfiniteScrollUtils.PageLoadCallback<DiscussionComment> callback) {
+            this.callback = callback;
+            if (isQuestionTypeThread && !isEndorsedFetched) {
+                getEndorsedList();
```

Comment:
Whoops, on a second look, I see that you are already have logic similar to what I described in (a), but also show the endorsed right when it's available. I guess that is the best user-experience, so... fine with me ;)